rust/Cargo.toml

```diff
@@ -4,6 +4,8 @@ members = [
   "connlib/clients/android",
   "connlib/clients/apple",
   "connlib/clients/shared",
+  "connlib/l4-tcp-dns-server",
+  "connlib/l4-udp-dns-server",
```

Comment:
My plan here is to also move the existing `dns-over-tcp` crate into the `connlib` directory and call it `l3-tcp-dns` because its API operates on layer 3 instead of layer 4.